### PR TITLE
Add Mint 17.1 (rebecca) support

### PR DESCRIPTION
**Pull request by WolverineFan**
Added the Mint 17.1 codename to the release lists.  Also adds LinuxMint support to apt::ppa (an updated version of my #361 pull request that resolves the add-apt-repository path problem)

### CODE REVIEW

**Comment by daenney** [comment]
I'm getting slightly annoyed by the amount of branches in our code specific to Mint...

**Comment by WolverineFan** [comment]
I'm inclined to agree.  But that's what they chose for developer desktops where I work so I figured I might as well submit the patch in case others need it.  It would be nice if Mint used the Ubuntu LSB strings.

**Comment by daenney** [comment]
Do you have any pull with Mint / know developers there that we could raise this with?

**Comment by WolverineFan** [comment]
No, I've submitted one patch and made a few comments/suggestions.  No influence to speak of.

File: manifests/backports.pp [hunk 1]
```diff
@@ -45,6 +45,7 @@
     } else {
       $distid = 'ubuntu'
       $release_real = $::lsbdistcodename ? {
+        'rebecca'  => 'trusty',
```
Comment (mhaskel) [inline manifests/backports.pp]: I know this makes the change bigger, but can you align the arrows in the selector?

File: manifests/params.pp [hunk 2]
```diff
@@ -17,6 +17,7 @@
       } else {
         $distid = 'ubuntu'
         $distcodename = $::lsbdistcodename ? {
+          'rebecca'  => 'trusty',
```
Comment (mhaskel) [inline manifests/params.pp]: Align arrows

File: manifests/ppa.pp [hunk 3]
```diff
@@ -14,8 +14,23 @@
...
-    fail('apt::ppa is currently supported on Ubuntu only.')
+  if $::operatingsystem == 'Ubuntu' {
+    $addaptrepository = '/usr/bin/add-apt-repository'
+  }
+  elsif $::operatingsystem == 'LinuxMint' {
+    # Prior to LinuxMint 17.1 (rebecca) the mintsources package installed a
+    # LinuxMint-specific add-apt-repository under /usr/local/bin.  As of 17.1 the
+    # software-properties-common package is no longer installed and mintsources
+    # puts add-apt-repository under /usr/bin
+    if $::lsbdistrelease > 17 {
```
Comment (mhaskel) [inline manifests/ppa.pp]: Can we use `versioncmp()` here instead?

**Comment by WolverineFan** [comment]
Incorporated the suggestions but haven't verified this on Mint 17.0 yet, so it's not ready yet.

**Comment by mhaskel** [comment]
@WolverineFan ok, great, thanks. It's looking good now, just ping us when it's verified.

**Comment by mhaskel** [comment]
@WolverineFan ping? any news on whether or not this worked for you for Rebecca?

**Comment by mhaskel** [comment]
Hi @WolverineFan, thanks for the contribution. For the 2.0.0 release we're removing mint parameters from params.pp, so I'm going to close this. Thanks again!

**Comment by WolverineFan** [comment]
What does that mean? Is Mint support being removed entirely or is there a more generic/portable solution?

**Comment by mhaskel** [comment]
@WolverineFan mint will still work, it's just removing the release string munging from parameters, so you'll need to pass the release parameter. It looks like we'll still need to update the new version of PPA to allow the command to be parameterized, but if you want to submit a new PR against 2.0.x with just that change that would be great, thanks!

**Comment by WolverineFan** [comment]
Excellent. Will do

**Pull request closed without merging**